rules/Naming/Rector/Class_/RenamePropertyToMatchTypeRector.php

```diff
@@ -139,12 +140,16 @@ private function refactorClassProperties(ClassLike $classLike): void
      * Such properties can have "xMock" names that are not compatible with "MockObject" suffix
      * They should be kept and handled by another naming rule that deals with mocks
      */
-    private function skipMockObjectProperty(Property $property): bool
+    private function skipDateTimeOrMockObjectPropertyType(Property $property): bool
     {
         if (! $property->type instanceof Name) {
             return false;
         }
 
-        return $this->isName($property->type, ClassName::MOCK_OBJECT);
+        if ($this->isName($property->type, ClassName::MOCK_OBJECT)) {
+            return true;
+        }
+
+        return $this->isName($property->type, DateTime::class);
```

Comment:
safe `DateTime` on string to avoid prefixed on downgrade

```suggestion
        return $this->isName($property->type, 'DateTime');
```

Reply:
:+1:

Reply:
https://github.com/rectorphp/rector-src/pull/6788/

Done :+1: